### PR TITLE
Patch/nsmarts finalize

### PR DESCRIPTION
**Pull request by johnmay**
A few more commits, a lot of these are quite small. Mostly should be self explanatory some stereochemistry corner cases needed resolving - I think these were broken in the old Parser too I just didn't notice.

I still need to add update documentation, particularly for ``QueryAtomContainer.create()`` which is 911c4b7, the utility of it is shown in b154fea.

### CODE REVIEW

**Comment by johnmay** [comment]
Documentation now added in last two commits.

File: base/isomorphism/src/main/java/org/openscience/cdk/isomorphism/QueryStereoFilter.java [hunk 1]
```diff
@@ -136,34 +136,50 @@ private boolean checkTetrahedral(int u, int[] mapping) {
...
+            int j = 0, k = 0;
+            for (; j < 4; j++) {
+                if (vs[i] == us[j])
+                    break;
+                if (us[j] == focusIdx)
+                    k = j;
+            }
+            if (j == 4)
+                us[k] = vs[i];
+        }
```
[the remark this answers is not in the document]
Comment (johnmay) [reply base/isomorphism/src/main/java/org/openscience/cdk/isomorphism/QueryStereoFilter.java]: I'll add some more comments, it's looking over two arrays and finding an item that doesn't exist and then replacing it with another one.

File: base/isomorphism/src/main/java/org/openscience/cdk/isomorphism/matchers/QueryAtomContainer.java [hunk 2]
```diff
@@ -1637,4 +1644,149 @@ public void setStereoElements(List<IStereoElement> elements) {
         this.stereoElements.clear();
         this.stereoElements.addAll(elements);
     }
+
+    public static QueryAtomContainer create(IAtomContainer mol, Expr.Type... opts) {
```
[the remark this answers is not in the document]
Comment (johnmay) [reply base/isomorphism/src/main/java/org/openscience/cdk/isomorphism/matchers/QueryAtomContainer.java]: Aren't our end-users developers? :-). Making substructure matches strict is very useful, for example ring bond count says I don't want to match large ring systems.

[the remark this answers is not in the document]
Comment (egonw) [reply base/isomorphism/src/main/java/org/openscience/cdk/isomorphism/matchers/QueryAtomContainer.java]: I mean end-users, like people using the CDK via KNIME, rcdk, etc, that take a list of SMILES and a list of SMARTS to do some screening... should people with those lists of SMARTS rewrite their queries in a different way? What advantage would that have? I was more thinking in that direction...

**Comment by johnmay** [comment]
Argh that broke the build... knew my algorithm was better :p

File: base/isomorphism/src/main/java/org/openscience/cdk/isomorphism/matchers/QueryAtomContainer.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (johnmay) [reply base/isomorphism/src/main/java/org/openscience/cdk/isomorphism/matchers/QueryAtomContainer.java]: I don't understand what you mean...? This have nothing to do with rewriting queries, this takes a 'real' molecule and turns it into a 'query' (ala QueryAtomContainerCreator). The SMARTS is still a SMARTS.

**Comment by egonw** [comment]
Good to go now?

**Pull request merged**